### DISCUSSION TITLE
In App Component

### DISCUSSION THREAD

**Discussion by vitorvlopes** [Telefonica/mistica-design#1574] [Proposals]
### Goal

The main objective of this discussion is to propose the implementation of a new component, called In-App, to improve the communication with app customers about important updates and changes. Through this discussion, we aim to:

- introduce the In-App component and its functionalities;
- discuss the different types of In-App and their use cases;
- identify the best practices for configuring and implementing the component;
- evaluate the impact of In-App on the design system and the user experience.

### Context

Effective communication with app users is crucial to ensure a positive user experience, and to keep customers informed about important updates, changes, and new features. With this in mind, we propose the implementation of a new component called In-App, based on the following features and functionalities:

- In-App component will be integrated into the existing CMS (Content Management System);
- The component will have four types of messages, which are message, localized, linked, and onboarding;
- The onboarding type will allow the user to be impacted by not only one message, but multiple consecutive messages.

### Impact

The use of In-App can have a significant impact on the user experience. Proper use of the component can improve the user experience by providing direct and effective communication with users, presenting important information about the application. This, in turn, can increase their engagement with the application.
![In App](https://github.com/Telefonica/mistica-design/assets/151069665/228824ea-e379-4d4f-8355-ae280fb2c104)
![In App Onboarding](https://github.com/Telefonica/mistica-design/assets/151069665/cbe34a6c-4d7b-4881-8610-207fb2a07cc9)

### Discussion Questions

- What are the best practices we can adopt to configure and use the different types of In-App?
- What needs to be taken into consideration when creating In-App content to ensure that it is relevant and interesting to the user?

### Do you have a visual proposal?

https://www.figma.com/file/rhYLXlDBEOI2muZGZ760bO/In-App?type=design&node-id=1%3A3&mode=design&t=MiqyLowOwIYcxPWq-1

**Comment by yceballost** [comment]
Thanks Vitor for the discussion

I dont understand what do you mean with "evaluate the impact of In-App on the design system and the user experience." 🤔 

I have some doubts about the image positioning, why do you think the image have to be place below the content? if you put the image at the bottom you will have misalignments when you scroll horizontally to the next page of the inapp. If you always place the image in the top, the next image will be aligned with the previous one avoiding "jumps". Jumps will appear when there are differences between the number of lines of two inapp pages.
